src/Microsoft.ML.StandardLearners/Standard/MultiClass/MetaMulticlassTrainer.cs

```diff
@@ -4,54 +4,76 @@
 
 using Float = System.Single;
 
+using System;
 using Microsoft.ML.Runtime.CommandLine;
 using Microsoft.ML.Runtime.Data;
 using Microsoft.ML.Runtime.Data.Conversion;
 using Microsoft.ML.Runtime.EntryPoints;
 using Microsoft.ML.Runtime.Internal.Calibration;
 using Microsoft.ML.Runtime.Internal.Internallearn;
 using Microsoft.ML.Runtime.Training;
+using Microsoft.ML.Core.Data;
 
 namespace Microsoft.ML.Runtime.Learners
 {
     using TScalarTrainer = ITrainer<IPredictorProducing<Float>>;
 
-    public abstract class MetaMulticlassTrainer<TPred, TArgs> : TrainerBase<TPred>
-        where TPred : IPredictor
-        where TArgs : MetaMulticlassTrainer<TPred, TArgs>.ArgumentsBase
+    public abstract class MetaMulticlassTrainer<TTransformer, TModel> : TrainerEstimatorBase<TTransformer, TModel>
+        where TTransformer : IPredictionTransformer<TModel>
+        where TModel : IPredictor
     {
         public abstract class ArgumentsBase
         {
-            [Argument(ArgumentType.Multiple, HelpText = "Base predictor", ShortName = "p", SortOrder = 1, SignatureType = typeof(SignatureBinaryClassifierTrainer))]
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Column to use for features", ShortName = "feat", SortOrder = 1)]
+            public string FeatureColumn = DefaultColumnNames.Features;
```

Comment:
>DefaultColumnNames.Features [](start = 42, length = 27)

If this is the design we are going for at least as a temporary measure we ought to have a flag on these to indicate that they should not appear either in entry-points or in the command line/GUI, since they will work contrary to the current mechanisms.

Reply:
However, I wonder, what is the purpose of having features and weights passed in for the case of OVA? It seems like something like OVA would take an estimator that decides how to train a model. This internal estimator would potentially know about weights and features, but the external structure seems like it only needs to know about the label so it knows how to "split up" the training.

---
In reply to: [216346589](https://github.com/dotnet/machinelearning/pull/865#discussion_r216346589) [](ancestors = 216346589)